src/pybind/mgr/cephadm/module.py

```diff
@@ -1248,27 +1279,25 @@ def _run_cephadm(self,
 
             self.log.debug('args: %s' % (' '.join(final_args)))
             if self.mode == 'root':
-                if stdin:
-                    self.log.debug('stdin: %s' % stdin)
-                script = 'injected_argv = ' + json.dumps(final_args) + '\n'
-                if stdin:
-                    script += 'injected_stdin = ' + json.dumps(stdin) + '\n'
-                script += self._cephadm
-                python = connr.choose_python()
-                if not python:
-                    raise RuntimeError(
-                        'unable to find python on %s (tried %s in %s)' % (
-                            host, remotes.PYTHONS, remotes.PATH))
                 try:
+                    cephadm_dest = CEPHADM_TOOL_PATH % (self._cluster_fsid, self.get_mgr_id())
+                    self._copy_cephadm(conn, cephadm_dest)
+
+                    self.log.info("Tying to execute : %s" % (['sudo', 'python', cephadm_dest] + final_args))
+                    if stdin:
+                        self.log.info("with stdin = %s" % stdin)
                     out, err, code = remoto.process.check(
                         conn,
-                        [python, '-u'],
-                        stdin=script.encode('utf-8'))
+                        ['sudo', 'python', cephadm_dest] + final_args,
```

Comment:
having 'python' hard coded here gave me issues on my machine(fedora30) because it only has python3 installed. so i got a 'command not found' error. could we still use the connr.choose_python() function here?

Reply:
How is that possible?, if you look at this line before the change it was called also "python" to execute the command. 
In which point are you having the issue? When you execute the bootstrap command? once you have the minimum cluster and start to add hosts? adding osds?.

Reply:
Good catch! You're completely right.

I hope we can get rid of this when we start working with a real executable :-)